uwblueprint/headsupguys
# Enabling Variable injection in MD Editor + Renderer

## Notion ticket link

[Kinda bypassed the ticket part but here's the relevant part of the feature spec](https://www.notion.so/uwblueprintexecs/Module-Builder-45d6b5fc94d24f039aa9ba2d91375499#51a4afeebf41486ebaab265325632eda)



## Implementation description
* 
- Markdown Editor now has a syringe which allows for injection of a variable
- - `<>[](variableName)`
- Markdown Renderer can display data if given data in the form shown above and if variables object is passed as a prop to it
- - If variables not passed as a prop a default message is displayed
- - If variables prop is passed but the given variable name is not a key found within, an empty string is displayed
- - If the key is found and the value is
- - - a string: injects the value as a string
- - - an array of strings: injects the value as an unordered list of strings



## Steps to test

1. Add/take away variables on the `editor.tsx` page and see if they display accurately in the renderer when called in the editor
2. Play around with the Markdown editor (specifically the syringe functionality) as well as adding styling

For example, if the variables object is `{key1: "value1", key2: ["a", "b", "c"]}`, one can display the variable data using `<>()[key1]` in the markdown editor


## What should reviewers focus on?
* Intuitiveness
* Any bugs


## Checklist
